lib/opencensus/absinthe/plug.ex

```diff
@@ -12,24 +12,19 @@ defmodule Opencensus.Absinthe.Plug do
     schema: MyApp.Schema,
     pipeline: {Opencensus.Absinthe.Plug, :traced_pipeline}
   ```
-
-  **WARNING:** `traced_pipeline/2` will be present _only_ if `Absinthe.Plug` is loaded.
-  Don't forget your `absinthe_plug` dependency in `mix.exs`!
   """
 
-  if Code.ensure_loaded?(Absinthe.Plug) do
```

Comment:
I copied the technique here from [`/apollo_tracing/pipeline.ex`](https://github.com/sikanhe/apollo-tracing-elixir/blob/58a91ea/lib/apollo_tracing/pipeline.ex#L8). (Not appealing to authority; just citing refs, and maybe they have some other mitigation against the race condition you mention.)

Is it appropriate to `rescue` … `e in UndefinedFunctionError` and `raise` a custom exception to let people know exactly what's going on? If not, maybe a note in the README they'll find while searching for `UndefinedFunctionError` while troubleshooting.

Reply:
I'm happy to hold off merging till it's clear the best way to proceed here.

In the original version of the code here https://github.com/sikanhe/apollo-tracing-elixir/blame/9fa7965f33c49913782b620d491b9100a56a607c/lib/apollo_tracing/pipeline.ex it was a runtime check. 

This https://github.com/sikanhe/apollo-tracing-elixir/pull/10#discussion_r157346314 is where it was changed so that library possibly has the bug since then. Though because it's _a_pollo and not _z_pollo it hasn't been much of a problem...

Reply:
Instead of removing this line you can add `absinthe_plug` as optional runtime dependency to force compilation of it before building `opencensus_absinthe`.

Reply:
:+1: I will do this when I next get chance, likely not for a week or two.

Do you know the reasoning behind not defining the function at all? Is it to be able to warn at compile time if it's used instead of erroring at runtime?

Reply:
I think they were trying to make it impossible to get a runtime error calling `Absinthe.Plug.default_pipeline/1`.